service/rds/api_op_CopyDBClusterSnapshot.go

```diff
@@ -378,3 +334,28 @@ func newServiceMetadataMiddleware_opCopyDBClusterSnapshot(region string) awsmidd
 		OperationName: "CopyDBClusterSnapshot",
 	}
 }
+
+// PresignCopyDBClusterSnapshot returns a Presigned HTTP request structure containing Request URL, method and SignedHeaders along with an error
+func (c *PresignClient) PresignCopyDBClusterSnapshot(ctx context.Context, input *CopyDBClusterSnapshotInput, optFns ...func(*Options)) (v4.PresignedHTTPRequest, error) {
```

Comment:
For S3 operations that can use the Expires custom header for presign URL expiry, `X-Amz-Expires`.

```go
presigner.PresignPutObject(ctx, input, func(o *rds.PresignOptions) {
     o.Presigner = v4.NewPresigner()
     o.OptionFns = append(o.OptionFns, func(o*Options){ o.Region = "us-west-2" })

     o.Expires = 20 * time.Minute // Only S3
})
````

Options for calling presign for an option
```go
PresignPutObject(context.Context, *PutObjectInput, ...func(*PresignOption))
````

Could have helpers for presign option with expires that uses type alias of the value set to not have a allocation from a function.

```go
func  WithPresignClientClientOptions(optFns ...func(*Options)) func(*PresignOptions) {
     return withPresignClientClientOptions(optFns).options
}
types withPresignClientClientOptions []func(*Options)
func (w withPresignClientClientOptions) option(o *PresignOptions) {
      o.ClientOptions = append(o.ClientOptions, w...)
}


func WithPresignExpires(dur time.Duration) func(*PresignOptions) {
      return withPresignExpires(dur).option
}
types withPresignExpires time.Duration
func (w withPresignExpires) option(o *PresignOptions) {
      o.Expires = time.Duration(w)
}